src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -179,19 +177,24 @@ internal virtual CustomAttributesBag<CSharpAttributeData> GetAttributesBag(Diagn
                 {
                     lazyAttributesStored = LoadAndValidateAttributes(
                         OneOrMany.Create(this.MergedAttributeDeclarationSyntaxLists),
-                        ref _lazyCustomAttributesBag,
-                        addToDiagnostics: diagnosticsOpt);
+                        ref _lazyCustomAttributesBag);
                 }
                 else
                 {
                     var typeParameter = (SourceTypeParameterSymbolBase)sourceMethod.SourcePartialDefinition.TypeParameters[_ordinal];
-                    CustomAttributesBag<CSharpAttributeData> attributesBag = typeParameter.GetAttributesBag(diagnosticsOpt);
+                    CustomAttributesBag<CSharpAttributeData> attributesBag = typeParameter.GetAttributesBag();
 
                     lazyAttributesStored = Interlocked.CompareExchange(ref _lazyCustomAttributesBag, attributesBag, null) == null;
                 }
 
                 if (lazyAttributesStored)
                 {
+                    if (ContainingSymbol is LocalFunctionSymbol localFunc)
+                    {
+                        var attributes = ((TypeParameterSyntax)_syntaxRefs[0].GetSyntax()).AttributeLists;
```

Comment:
>[0] [](start = 74, length = 3)

Use ```Single()``` instead because you are making an assumption that there is only one syntax reference.